### PR TITLE
Issue #40 : Organization FertilizerInspection Schema

### LINKED ISSUE #40: As a dev, I want to block the duplication/spreading of data between the manufacturer and company sections.
## Description  
Right now the application mixes or duplicates the information in the "company" and "manufacturer" fields. This leads to confusion and inaccuracies in the data presented to the users. To resolve this, a simple fix can be implemented to manage the results obtained by GPT on the backend.

**Bug Summary:** 
The backend returns results that often duplicate or mix the information in the "company" and "manufacturer" fields. This causes redundancy and data inconsistency.

**Steps to Reproduce:**  
1. Login into the app.
2. Analyse picture give in this issue or that contain multiple place where the same information of compagny is wrote.
3. Observe the returned results in the respective fields.

**Expected Behavior:**  
The information in the "company" and "manufacturer" fields should be distinct without duplication or mixing. Each field should contain only relevant information.

**Actual Behavior:**  
The "company" and "manufacturer" fields often contain duplicated or mixed information, causing confusion and redundancy.

**Environment:**  

**Commit Hash:** [e.g., 1cb3d1876a7b4c2435b4a5067e5ae53631063948]  

**OS Version:** Windows 10

**Browser Version:** Chrome Version: 129.0.6668.59 (Build officiel) (64 bits)

**Logs and Screenshots:**  
[Follow this link to the folder data](https://136gc.sharepoint.com/:f:/s/CFIAAILabTeam-harvester/EjbhYOYx5AdMs-TaRyMCoqkBScyypC9U42fao-VV8-lwbQ?e=mGnOlH)

**Suggested Fix:** (facultatif)  
Manage the results obtained by GPT on the backend. Currently, the backend returns a result almost identical to the frontend. A simple string matching algorithm could be used to check if the strings are in multiple field. If so, everything should be merged into the "company" for easier user compliance.

### PR DESCRIPTION
**Pull request by snakedye**
Replaces `manufacter_` and `company_` by an `Organization` object.

Closes #40

### CODE REVIEW

**Comment by k-allagbe** [comment]
How does it react with real labels?

**Comment by snakedye** [comment]
@k-allagbe 
Seems to work well. I did run the pipeline on ten labels and in all of them at least one organizations was picked up.

## Example 1

image

```json
{
  "organizations": [
    {
      "name": "TerraLink Horticulture Inc.",
      "address": "464 Riverside Road Abbotsford, BC V25 7M1",
      "website": "www.tlhort.com",
      "phone_number": null
    }
  ],
  "fertiliser_name": "Granular Fertilizers for Professional Growers",
  ...
}
```

## Example 2

img_001

```json
{
  "organizations": [
    {
      "name": "Diamond Fertilizers Inc.",
      "address": "PO Box 5508 Stn Main High River, AB CANADA T1V 1M6",
      "website": "www.soil-aid.com",
      "phone_number": "+18663372943"
    }
  ],
  "fertiliser_name": "Super Crop® Foliar Booster",
  ...
}
```

## Example 3

img_001 (1)

```json
{
  "organizations": [
    {
      "name": "Groupe 3MR Inc.",
      "address": "Ampère, bureau 200. Boucherville (Québec) JAB 526",
      "website": null,
      "phone_number": null
    }
  ],
  "fertiliser_name": "Botaflora",
  ...
}
```

## Example 4

img_001 (2)

```json
{
  "organizations": [
    {
      "name": "Advanced Micronutrient Products, Inc.",
      "address": "2405 WAPassar Rd. Reese, MI-18.757-USA",
      "website": null,
      "phone_number": null
    }
  ],
  "fertiliser_name": "Micronutrient Corn Mix Mélange d'oligoéléments pour maïs",
  ...
}
```

File: tests/test_pipeline.py [hunk 1]
```diff
@@ -195,7 +195,7 @@ def test_label_006_website_inspection(self):
         inspection = analyze(label_storage, self.ocr, self.gpt)
 
         # Assertions for website fields
-        self.assertEqual(inspection.company_website, "www.activeagriscience.com")
+        self.assertIn("activeagriscience.com", str(inspection.organizations))
```
Comment (k-allagbe) [inline tests/test_pipeline.py:198]: This was done as per requirements for #60. Please try with a prompt similar to the old one. If not working, we'll do it in the validation function.

File: pipeline/inspection.py [hunk 2]
```diff
@@ -23,6 +22,38 @@ def extract_first_number(string: str) -> Optional[str]:
...
 
+class Organization(BaseModel):
+    """
+    Represents an organization such as a manufacturer, company, or any entity 
+    associated with a fertilizer.
+    """
+    name: Optional[str] = Field(None, description="The name of the organization.")
+    address: Optional[str] = Field(None, description="The address of the organization.")
+    website: Optional[str] = Field(None, description="The website of the organization.")
+    phone_number: Optional[str] = Field(None, description="The phone number of the organization.")
```
Comment (k-allagbe) [inline pipeline/inspection.py:33]: Please try a prompt similar to the old one see if it picks up phone numbers in test labels 22, 24, 28 & 34.

**Review by k-allagbe — APPROVED** [review]

**Pull request merged**